internal/core/unittest/test_utils/DataGen.h

```diff
@@ -317,14 +326,95 @@ GenerateRandomSparseFloatVector(size_t rows,
     return tensor;
 }
 
-inline GeneratedData DataGen(SchemaPtr schema,
-                             int64_t N,
-                             uint64_t seed = 42,
-                             uint64_t ts_offset = 0,
-                             int repeat_count = 1,
-                             int array_len = 10,
-                             bool random_pk = false,
-                             bool random_val = true) {
+inline void generateRandomPoint(OGRPoint& point) {
+    point.setX(static_cast<double>(rand()) / RAND_MAX * 360.0 - 180.0);
+    point.setY(static_cast<double>(rand()) / RAND_MAX * 180.0 - 90.0);
+}
+
+inline void
+generateRandomLineString(OGRLineString& lineString, int numPoints) {
+    for (int i = 0; i < numPoints; ++i) {
+        OGRPoint point;
+        generateRandomPoint(point);
+        lineString.addPoint(&point);
+    }
+}
+
+inline void
+generateRandomPolygon(OGRPolygon& polygon, int numperRing) {
+    OGRLinearRing ring;
+    generateRandomLineString(ring, numperRing);
+    ring.closeRings();
+    polygon.addRing(&ring);
+}
+
+inline OGRGeometry*
+GenRandomGeometry() {
+    OGRGeometry* geometry = nullptr;
+    int geomType = rand() % 6;  // Randomly select a geometry type (0 to 6)
+    switch (geomType) {
+        case 0: {
+            OGRPoint point;
+            generateRandomPoint(point);
+            geometry = point.clone();
+            break;
+        }
+        case 1: {
+            OGRLineString lineString;
+            generateRandomLineString(lineString, 5);
+            geometry = lineString.clone();
+            break;
+        }
+        case 2: {
+            OGRPolygon polygon;
+            generateRandomPolygon(polygon, 5);
+            geometry = polygon.clone();
+            break;
+        }
+        case 3: {
+            OGRMultiPoint multiPoint;
+            for (int i = 0; i < 3; ++i) {
+                OGRPoint point;
+                generateRandomPoint(point);
+                multiPoint.addGeometry(&point);
+            }
+            geometry = multiPoint.clone();
+            break;
+        }
+        case 4: {
+            OGRMultiLineString multiLineString;
+            for (int i = 0; i < 3; ++i) {
+                OGRLineString lineString;
+                generateRandomLineString(lineString, 5);
+                multiLineString.addGeometry(&lineString);
+            }
+            geometry = multiLineString.clone();
+            break;
+        }
+        case 5: {
+            OGRMultiPolygon multiPolygon;
+            for (int i = 0; i < 3; ++i) {
+                OGRPolygon polygon;
+                generateRandomPolygon(polygon, 5);
+                multiPolygon.addGeometry(&polygon);
+            }
+            geometry = multiPolygon.clone();
+            break;
+        }
+    }
+    return geometry;
+}
+
+inline GeneratedData
```

Comment:
The order of function DataGen is changed, the parameter 'repeat_count' become the 6th parameter, this will lead to the failures of test_group_by, which pass a non-zero repeat_count but got a zero repeat count after this change. Please modify the corresponding ut case to avoid Signal 8 (SIGFPE)